Warehouse.Application/Features/Commands/Products/ProductDelete/ProductDeleteCommandHandler.cs

```diff
@@ -8,14 +8,25 @@ public class ProductDeleteCommandHandler(IUnitOfWork _unitOfWork) : IRequestHand
 {
     public async Task Handle(ProductDeleteCommand command, CancellationToken cancellationToken)
     {
-        var existingProduct = await _unitOfWork.Products.GetById(command.productId);
-        if (existingProduct is null)
+        try
         {
-            throw new ProductNotFoundException($"Product with ID {command.productId} not found.");
-        }
+            var existingProduct = await _unitOfWork.Products.GetById(command.productId);
+            if (existingProduct is null)
+            {
+                throw new ProductNotFoundException($"Product with ID {command.productId} not found.");
+            }
 
-        existingProduct.IsDeleted = true;
-        _unitOfWork.Products.Update(existingProduct);
-        await _unitOfWork.SaveAsync();
+            _unitOfWork.Products.Delete(existingProduct);
+            _unitOfWork.Commit();
+        }
+        catch (Exception ex)
+        {
+            _unitOfWork.Rollback();
+            throw new ApplicationException("An error occurred while processing the request.", ex);
+        }
+        finally
```

Comment:
wouldn't this construction be better:
try/catch/finally be part of the unit of work's commit method so we don't need to put it in every handler.
What do you think of as an approach?